### ISSUE TITLE
function indentation linter

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2003]
Would be nice to have a linter encouraging use of the new double-indented function declaration style where appropriate:

https://style.tidyverse.org/functions.html#long-lines-1

A key is:

> Prefer function-indent style to double-indent style when it fits.

Here's a work-in-progress XPath for the linter:

```xpath
//FUNCTION
  /parent::expr[OP-LEFT-PAREN/@line1 = SYMBOL_FORMALS[1]/@line1]
  /EQ_FORMALS[@line1 != following-sibling::expr[1]/@line2]
  /following-sibling::expr[1]
```

Followed by `nchar(xml2::xml_text(xml)) < 74L`, where `74` is `80` - `4` (double indent) - `2` (this expression can be bumped to the next line). `<` is to allow `,` after the expression.

It needs some fine-tuning:

 - It should only apply to assignments (currently anonymous functions would be linted)
 - It should skip comments. _a la_ `xml2lang()`.
 - We should generalize with parameters `max_line_length` and `indentation`, then the filter is `nchar(.) < max_line_length  - 3 * indent_spaces`
 - Terminal arguments have a slightly different constraint since they're followed by `) {`, not `,`.

We could also (optionally?) always lint multiline function definitions like

```r
foo <-
  function(...) { ... }
```

**Comment by MichaelChirico** [comment]
Not sure this is relevant after the updated ruling on function indentation, which doesn't mention a preference between the two mutli-line definition options.
